### PR TITLE
chore: Update documentation to clarify relationship queries

### PR DESCRIPTION
**Pull request by iamber12**
This PR refines the documentation to provide clearer guidance on handling fields related through parent-child or lookup relationships when querying Salesforce objects. The adjustments ensure that users understand how to structure their queries and corresponding Go types effectively.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,6 +186,29 @@ if err != nil {
 }
 ```
 
+### Handling Relationship Queries
+
+When querying Salesforce objects, it's common to access fields that are related through parent-child or lookup relationships. For instance, querying `Account.Name` with related `Contact` might look like this:
+
+##### Example SOQL Query
+```sql
```
Comment (ccoVeille) [inline README.md:194]:
```suggestion

```sql
```

File: README.md [hunk 2]
```diff
@@ -186,6 +186,29 @@ if err != nil {
...
+SELECT Id, Account.Name FROM Contact
+```
+
+#### Corresponding Go Structs
+To effectively handle the data returned by this query, define your Go structs as follows:
```
Comment (ccoVeille) [inline README.md:199]:
```suggestion

To effectively handle the data returned by this query, define your Go structs as follows:
```

File: README.md [hunk 3]
```diff
@@ -186,6 +186,29 @@ if err != nil {
...
+
+```go
+type ContentDocumentLink struct {
+    Id       string
+    Account Account
+}
+
+type Account struct {
+    Name string
```
Comment (ccoVeille) [inline README.md:208]: Don't you have a tag thing to define?


```suggestion
    Name string `soql:"name"`
```


This could be useful for Salesforce formula fields that have strange name, it may be addressed with another PR if it doesn't exist yet

Comment (k-capehart) [reply README.md:208]: @ccoVeille Only if you are using the `QueryStruct()` method, but that's not necessary for this example.

**Review by k-capehart — APPROVED** [review]
Looks good to me! Thanks @iamber12
